### PR TITLE
stylo: Implement dirtyness tracking for stylo.

### PR DESCRIPTION
**Pull request by emilio**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because geckolib-only changes.

r? @bholley

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12339)

### CODE REVIEW

**Comment by emilio** [comment]
Bobby is on PTO, so...

r? @heycam

File: ports/geckolib/gecko_bindings/structs_debug.rs [hunk 1]
```diff
@@ -2760,6 +2764,84 @@ pub enum SheetParsingMode {
...
+pub enum DOMPoint { }
+pub enum DOMQuad { }
+pub enum DOMRectReadOnly { }
+pub enum Element { }
+pub enum Text { }
+pub enum TextOrElementOrDocument { }
+pub enum DOMPointInit { }
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum NodeFlags {
```
Comment (heycam) [inline ports/geckolib/gecko_bindings/structs_debug.rs]: I'm curious, what causes the name NodeFlags to be generated here, given the enum in nsINode.h has no name?

**Comment by aneeshusa** [comment]
@bors-servo retry

- Very strange permissions in the `arm64` build directory on the `servo-linux-cross` builder: looks like these files were owned by `libuuid/libuuid` (UID 100, GID 101). Not sure how that happened...

**Pull request merged**